pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -164,19 +164,12 @@ type resettableRESTMapper interface {
 // the mapper's underlying discovery client will be unnecessarily reset during
 // the course of detecting new resources.
 func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterface, period time.Duration, stopCh <-chan struct{}) {
+	var err error
 	oldResources := make(map[schema.GroupVersionResource]struct{})
 	wait.Until(func() {
 		// Get the current resource list from discovery.
 		newResources := GetDeletableResources(discoveryClient)
 
-		// This can occur if there is an internal error in GetDeletableResources.
-		// If the gc attempts to sync with 0 resources it will block forever.
-		// TODO: Implement a more complete solution for the garbage collector hanging.
```

Comment:
Removing this TODO and the code below because this PR implements the more complete solution